bridges/snowbridge/pallets/outbound-queue-v2/src/lib.rs

```diff
@@ -417,20 +420,26 @@ pub mod pallet {
 		}
 
 		/// Process a delivery receipt from a relayer, to allocate the relayer reward.
-		pub fn process_delivery_receipt(
-			relayer: <T as frame_system::Config>::AccountId,
-			receipt: DeliveryReceipt,
-		) -> DispatchResult {
+		pub fn process_delivery_receipt(receipt: DeliveryReceipt) -> DispatchResult
+		where
+			<T as frame_system::Config>::AccountId: From<[u8; 32]>
+		{
 			// Verify that the message was submitted from the known Gateway contract
 			ensure!(T::GatewayAddress::get() == receipt.gateway, Error::<T>::InvalidGateway);
 
+			let reward_account: T::AccountId = receipt.reward_address.into();
```

Comment:
As a fallback, if `receipt.reward_address` is zero bytes, we should rather reward the relayer.

Reply:
one comment here: https://github.com/claravanstaden/polkadot-sdk/commit/121341aebde3b3d78609e16906a430a1cc516714#comments

Reply:
Comment link: https://github.com/claravanstaden/polkadot-sdk/commit/121341aebde3b3d78609e16906a430a1cc516714#r155493682

Reply:
https://github.com/claravanstaden/polkadot-sdk/pull/20/commits/2c493f1770b5f4591067613b633ee2e3883585bb